servicediscovery/mdns/mdns.go

```diff
@@ -24,28 +24,34 @@ type resolver struct {
 }
 
 func (z *resolver) ResolveID(req servicediscovery.ResolveRequest) (string, error) {
-	port, err := LookupPortMDNS(req.ID)
+	address, err := lookupAddressMDNS(req.ID)
 	if err != nil {
 		return "", err
 	}
-	return fmt.Sprintf("localhost:%v", port), nil
+	return address, nil
 }
 
-// LookupPortMDNS uses mdns to find the port of a given service entry on a local network
-func LookupPortMDNS(id string) (int, error) {
+// lookupAddressMDNS uses mdns to find the port of a given service entry on a local network
+func lookupAddressMDNS(id string) (string, error) {
 	resolver, err := zeroconf.NewResolver(nil)
 	if err != nil {
-		return -1, fmt.Errorf("failed to initialize resolver: %e", err)
+		return "", fmt.Errorf("failed to initialize resolver: %e", err)
 	}
 
 	port := -1
+	ip := "localhost" // default
```

Comment:
will `ip` be overwritten all the time ?

Reply:
I would expect it to be yes but this was just a safety in case the entry didn’t have a registered IPv4 or IPv6 address for some reason. I will add an additional condition to the entry for a default case and leave this uninitialized to reduce alloc.